state/src/storage/tx/raft_log.rs

```diff
@@ -233,13 +234,11 @@ impl<'db> StateTxn<'db, RW> {
 
     /// Purge all logs before the given index (inclusive)
     pub fn purge_logs(&self, index: u64) -> Result<(), StorageError> {
-        let idx_lsn = lsn_to_key(index);
-        let first_log = self.first_raft_log_index()?.unwrap_or(0);
-
-        // Purge the logs
         let mut log_cursor = self.logs_cursor()?;
+
+        let idx_lsn = lsn_to_key(index);
         log_cursor.seek(&idx_lsn)?;
-        for _ in first_log..=index {
+        while log_cursor.has_current()? {
             log_cursor.delete()?;
```

Comment:
just want to confirm that `log_cursor.delete()` will not progress the cursor's position, right?

Reply:
it _does_ move the cursor, this is the desired behavior and there is a test for this